src/lib/blockstore.js

```diff
@@ -13,7 +13,8 @@ import { BlockBatch } from './block-batch.js'
  * @typedef {import('../bindings').R2Bucket} R2Bucket
  */
 
-const MAX_BLOCK_LENGTH = 1024 * 1024 * 4
+// 2MB (max safe libp2p block size) + typical block header length + some leeway
+const MAX_ENCODED_BLOCK_LENGTH = (1024 * 1024 * 2) + 39 + 61
```

Comment:
We don't know how much to read from the CAR, but in theory no single block should be bigger than this.